Makefile

```diff
@@ -71,26 +78,14 @@ lint-go:
 lint: ensure-submodule-initialized lint-cpp lint-go
 
 setup: ensure-submodule-initialized
-ifeq ("Darwin", "$(shell uname -s)")
-	cd viam-cartographer/scripts && ./setup_cartographer_macos.sh
-else
-	cd viam-cartographer/scripts && ./setup_cartographer_linux.sh
-endif
+	viam-cartographer/scripts/setup_cartographer.sh
 	@make artifact-pull
-	
-build: build-module
-ifneq ($(wildcard viam-cartographer/cartographer/build/.),)
-	cd viam-cartographer && ./scripts/build_viam_cartographer.sh 
-else 
-	cd viam-cartographer && ./scripts/build_cartographer.sh && ./scripts/build_viam_cartographer.sh
-endif
 
-build-debug: build-module
-ifneq ($(wildcard viam-cartographer/cartographer/build/.),)
-	cd viam-cartographer && ./scripts/build_viam_cartographer_debug.sh 
-else 
-	cd viam-cartographer && ./scripts/build_cartographer.sh && ./scripts/build_viam_cartographer_debug.sh
-endif
+build: ensure-submodule-initialized buf build-module
+	cd viam-cartographer && cmake -Bbuild -G Ninja ${EXTRA_CMAKE_FLAGS} && cmake --build build
+
+build-debug: EXTRA_CMAKE_FLAGS+=" -DCMAKE_BUILD_TYPE=Debug"
+build-debug: build
```

Comment:
Can we combine these?

Reply:
I had noted on line 82 that we coudl run make commands on separate lines using a @make line first. However, I just ran a test and, no luck.

Reply:
I was mistaken about suppression. The @ symbol only suppresses the echoing of the command, not the output. That said, I still think it's better to use true dependencies rather than sub-make processes (which are effectively just calling make again from a new shell.)